networking/Field_Control_Networking_Spec.md

```diff
@@ -0,0 +1,83 @@
+
+## 2. Setting Up the Clients and Network
+
+### Clients
+
+The FC network will have up to eight clients, as follows:
+
+- One FC client.  This is the FC computer running the match.  A PiE computer+monitor. 
+- Four driver station clients.  This acts as the student-robot interface (via Dawn and Runtime), and as the robot-field control interface (see "Establishing Communication Links").  A wifi-enabled laptop or  computer+monitor.
+- Up to four robot clients.  A wifi-enabled Beagle Bone Black (via wifi dongle).  One per student robot.
+
+### Assigning IP Addresses
+
+Every client will have a statically assigned IP address.  This is feasible due to the small number of total clients (1 per robot ( less than 30 in total), 4 driver stations, and one field control center).
+
+Consider the following example static IP scheme ("P" = subnet prefix):
+
+- FCl:  P.255
+- Driver Stations:  P.254-251
+- Robots:  P.TeamNumber
+
+## 3. Connecting to the Field Control Network
+
+### Configuration
+
+A robot can connect to at most one network at a time, which must be either the Student network or the FC network.  The Student network is to be listed as higher priority than the FC Network.  This can be set each robot's network config file.
+
+In order to participate in a match, teams temporarily relinquish their routers.
+
+### Rationale 
+
+This results in two desirable guarantees:
+
+1. When not in a match, students will always be able to connect with their robots over the Student network even if the robot is within range of the FC network.
+2. When in a match, robots will only have the option of connecting to the FC network (due to router confiscation). 
+
+Note that this may result in robots not in a match to still be connected to the FC network.  However, this is totally fine since the FC client will not be listening to robots not currently tied to a driver station (see "Sending Messages").
+
+## 4. Establishing Communication Links
+
+Within each match, the FC network will create/maintain eight TCP connections:
+
+- A connection between the FC client and each Driver Station client.  These can persist across matches, and so need to be opened only once per network initialization.
+- A connection beween a Driver Station client and its corresponding Robot client. These must be opened/closed at the start/end of each match (i.e. once the Robot client joins/leaves the FC network and is paired/unpaired with a Driver Station client).
+
+If a FC-to-Driver-Station connection is dropped, FC is responsible for reestablishing the connection.  If a Driver-Station-to-Robot connection is dropped, the Driver Station notifies FC; FC then re-sends the pairing message to said Driver Station (see "Sending Messages").
+
+## 5. Sending Messages
+
+The FC client sends messages to initiate robot-to-driver-station pairing, and to indicate what mode (autonomous, tele-op, etc) the robot is in.   Driver station clients listen accordingly, and pass the message to robots.
```

Comment:
robot should initiate pairing via message to FC as it is generally unclear when it is ready to communicate unless inspecting every packet